### PR TITLE
Feat/book selected reservations

### PR DESCRIPTION
**Pull request by GGmaz**
Client can add items to cart, remove them or change their quantity. After that, he can shop and empty its cart

### CODE REVIEW

File: src/main/java/com/example/cartservice/serviceImpl/CartServiceImpl.java [hunk 1]
```diff
@@ -1,14 +1,80 @@
...
+    public CartServiceImpl(CartRepository cartRepository, ReservationRepository reservationRepository, InvoiceRepository invoiceRepository) {
+        this.cartRepository = cartRepository;
+        this.reservationRepository = reservationRepository;
+        this.invoiceRepository = invoiceRepository;
+    }
+
+
+    @Override
+    public ResponseStatusException shop(Long cartId) {
+        Invoice invoice = new Invoice();
```
Comment (milosevm) [inline src/main/java/com/example/cartservice/serviceImpl/CartServiceImpl.java:37]: Neki exception handler bi bio bas super ovde

File: src/main/java/com/example/cartservice/serviceImpl/ReservationServiceImpl.java [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
 @Service
 public class ReservationServiceImpl implements ReservationService {
 
     @Autowired
     private ReservationRepository reservationRepository;
+
+
+    @Override
+    public boolean isHotelAvailable(ReservationDTO resDto) {
+        Integer occupied = 0;
```
Comment (milosevm) [inline src/main/java/com/example/cartservice/serviceImpl/ReservationServiceImpl.java:21]: Gledajte da uvek vracate ResponseEntity umesto svojih objekata ili prostih tipova.

File: src/main/resources/data.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO CART(id, client_id) values (1, 1);
```
Comment (milosevm) [inline src/main/resources/data.sql:1]: Gledajte da napravite seeder klase

File: src/main/java/com/example/cartservice/serviceImpl/ReservationServiceImpl.java (hunk 2 above)
Comment (GGmaz) [reply src/main/java/com/example/cartservice/serviceImpl/ReservationServiceImpl.java:21]: Bas iz servisa da vracam ResponseEntity?

**Review by milosevm — APPROVED** [review]

**Pull request merged**